src/createRollup.ts

```diff
@@ -180,16 +180,19 @@ export async function createRollup({
     serializedTransaction: await account.signTransaction(txRequest),
   });
 
-  // get the transaction
-  const tx = createRollupPrepareTransaction(
-    await validatedParentChainPublicClient.getTransaction({ hash: txHash }),
-  );
-
   // get the transaction receipt after waiting for the transaction to complete
   const txReceipt = createRollupPrepareTransactionReceipt(
     await validatedParentChainPublicClient.waitForTransactionReceipt({ hash: txHash }),
   );
 
+  // get the transaction
+  // (we get the transaction after waiting for the receipt, to prevent `TransactionNotFoundError`s coming
+  // from RPCs that use load balancing and caching. More information can be found here:
+  // https://github.com/wevm/viem/issues/1056#issuecomment-1689800265 )
+  const tx = createRollupPrepareTransaction(
+    await validatedParentChainPublicClient.getTransaction({ hash: txHash }),
+  );
```

Comment:
This is needed to prevent cases where the RPC fails to find the transaction that was just sent (due to load balancing or caching). For example, when using the public ArbSepolia RPC, calling this method fails when requesting the transaction. More information here: https://github.com/wevm/viem/issues/1056#issuecomment-1689800265